Modifies the Makefile to work with OpenEmbedded.

I copied the CFLAGS_R approach from the existing 2.0 squashfs-tools, 
without actually knowing why this approach was taken. It works though.

Leon Woestenberg <leonw@mailcan.com>

--- squashfs-tools/Makefile	2006-10-10 13:59:05.000000000 +0200
+++ squashfs-tools/Makefile.oe	2006-10-10 14:06:01.000000000 +0200
@@ -1,17 +1,22 @@
 INCLUDEDIR = .
+CFLAGS_R = -I$(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+LIBS = -lz -lpthread
 
 CFLAGS := -I$(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -O2
 
 all: mksquashfs unsquashfs
 
 mksquashfs: mksquashfs.o read_fs.o sort.o
-	$(CC) mksquashfs.o read_fs.o sort.o -lz -lpthread -o $@
+	$(CC) $(LDFLAGS) mksquashfs.o read_fs.o sort.o $(LIBS) -o $@
 
 mksquashfs.o: mksquashfs.c squashfs_fs.h mksquashfs.h global.h sort.h
+	$(CC) $(CFLAGS_R) $(CFLAGS) -c -o mksquashfs.o mksquashfs.c
 
 read_fs.o: read_fs.c squashfs_fs.h read_fs.h global.h
-
+	$(CC) $(CFLAGS_R) $(CFLAGS) -c -o read_fs.o read_fs.c
+	
 sort.o: sort.c squashfs_fs.h global.h sort.h
+	$(CC) $(CFLAGS_R) $(CFLAGS) -c -o sort.o sort.c
 
 unsquashfs: unsquashfs.o
 	$(CC) unsquashfs.o -lz -o $@
